File: src/main/kotlin/net/starlegacy/feature/gas/collectionfactors/RandomFactor.java
package net.starlegacy.feature.gas.collectionfactors;

import org.bukkit.Location;

public class RandomFactor extends CollectionFactor {

    private float chance;

    public RandomFactor(float chance) {
        this.chance = chance;
    }

    @Override
    public boolean factor(Location location) {
        return Math.random() <= chance;
    }
}

File: server/src/main/kotlin/net/horizonsend/ion/server/features/gas/collectionfactors/RandomFactor.kt
package net.horizonsend.ion.server.features.gas.collectionfactors

import org.bukkit.Location

open class RandomFactor(private val chance: Float) : CollectionFactor() {
    override fun factor(location: Location): Boolean {
        return Math.random() <= chance
    }

	override fun canBeFound(location: Location): Boolean = true
}
